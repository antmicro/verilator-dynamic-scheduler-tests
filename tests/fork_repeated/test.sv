module t;

  int count = 0;

  function void f();
    fork
      begin
        $write("forked process\n");
        count++;
      end
      begin
        $write("forked process\n");
        count++;
      end
      begin
        $write("forked process\n");
        count++;
      end
    join_none
  endfunction
  initial begin    
    f();
    f();
    f();
    wait(count == 9);
    $write("*-* All Finished *-*\n");
    $finish();
  end
endmodule
