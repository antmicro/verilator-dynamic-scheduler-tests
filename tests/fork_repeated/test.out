forked process
forked process
forked process
forked process
forked process
forked process
forked process
forked process
forked process
*-* All Finished *-*
